modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -481,7 +488,32 @@ using GConstGIEModel = ade::ConstTypedGraph
     , IECallable
     >;
 
+inline IE::Blob::Ptr extractRemoteBlob(IECallContext& ctx, std::size_t i) {
+    GAPI_Assert(ctx.inShape(i) == cv::GShape::GFRAME &&
+                        "Callback feature is supported for MediaFrame only");
+
+    cv::util::any any_blob_params = ctx.inFrame(i).blobParams();
+    auto ie_core = cv::gimpl::ie::wrap::getCore();
+
+    using ParamType = std::pair<InferenceEngine::TensorDesc,
+                                InferenceEngine::ParamMap>;
+
+    ParamType* blob_params = cv::util::any_cast<ParamType>(&any_blob_params);
+    if (blob_params == nullptr) {
+        GAPI_Assert(false && "Incorrect type of blobParams: "
+                              "expected std::pair<InferenceEngine::TensorDesc,"
+                                                 "InferenceEngine::ParamMap>");
+    }
+
+    return ctx.uu.params.rctx->CreateBlob(blob_params->first,
+                                          blob_params->second);
+}
+
 inline IE::Blob::Ptr extractBlob(IECallContext& ctx, std::size_t i) {
+    if (ctx.uu.params.rctx != nullptr) {
+        return extractRemoteBlob(ctx, i);
+    }
```

Comment:
See above - this is not robust.

Imagine the user has configured pipeline once (including passing the remote context) but then has used our host-side `GCaptureSource` for input. There will be no any remote params in frames produced by that - and it is a perfectly valid situation.

Reply:
I don't agree. If user sets a config for remote context and it's successfully created, it then imports/loads network using this remote context. I don't think they will be valid for non-remote usage (e. g. if we are just creating blobs on host). If the users sets the config by mistake - they will see this error and remove it.

Reply:
We need to document this behavior somewhere clearly

Reply:
Added some documentation on `cfgContextParams`

Reply:
A simple NB within `giebackend.cpp` (here at this point) would also work